File: S1/HMIN108 - Programmation orientee agents/TDs-TPs/2017-2018/Warbot/teams/python_teamFunction_level_0/WarExplorer.py

def actionWarExplorer():

    percepts = getPercepts();

    for percept in percepts:
        if(percept.getType().equals(WarAgentType.WarFood)):
            if((percept.getDistance() < getMaxDistanceTakeFood()) and (not isBagFull())):
                setHeading(percept.getAngle());
                return take();
            elif (not isBagFull()) :
                setHeading(percept.getAngle());

    if (isBagFull()) :
        setDebugString("Bag full return base")

        percepts = getPerceptsAlliesByType(WarAgentType.WarBase);

        if((percepts is None) or (len(percepts) == 0)):
            #broadcastMessageToAll("whereAreYou", "");

            messages = getMessages();

            for message in messages :
                if(message.getSenderType() == WarAgentType.WarBase):
                    setHeading(message.getAngle());

            broadcastMessageToAgentType(WarAgentType.WarBase, "whereAreYou", "");

        else :
            base = percepts[0];

            if(base.getDistance() > maxDistanceGive()):
                setHeading(base.getAngle());
                return move();
            else:
                setIdNextAgentToGive(base.getID());
                return give();
    else :
        setDebugString("Chercher food");

    if (isBlocked()) :
        RandomHeading()

    return move()

File: S1/HMIN108 - Programmation orientee agents/TDs-TPs/TP6 - Warbot/warbot-3.3.3/teams/javascript_level_0/WarExplorer.js
var Explorer = JsWarExplorer.extend({

	action : function() { 

		percepts = this.getPercepts();

		var baseEnum = edu.warbot.agents.enums.WarAgentType.WarBase;

		//println(baseEnum);

		for (i = 0; i < percepts.size(); i++) { 
			//println("type percept");
			//println(percepts.get(i).getType());
			//println(this.getFoodType());
			if(percepts.get(i).getType() == this.getFoodType()) {
				//println("food view");
				if((percepts.get(i).getDistance() < this.getMaxDistanceTakeFood()) && (!this.isBagFull())) {
					this.setHeading(percepts.get(i).getAngle());
					return this.take();
				}
				else if (!this.isBagFull()) {
					this.setHeading(percepts.get(i).getAngle());
				}
			}
		}

		if(this.isBagFull()) {
			this.setDebugString("Bag full return base");

			/*perceptsBase = null;

			for (j = 0; j < percepts.size(); j++) { 
				if(percepts.get(i).getType == "WarBase") {
					perceptsBase = percepts.get(i);
				}
			}
			
			

			if(perceptsBase == null) {
				messageAgent = this.getMessages();
				mes = null;

				if(messageAgent != null) {
					for (i = 0; i < messageAgent.size(); i++) {
						mes = messageAgent.get(i);
						if(mes.getSenderType() == "WarBase") {
							mes = null;
						}
					} 
				}
				if(mes != null) {
					this.setHeading(mes.getAngle());
				}

				this.broadcastMessageToAgentType(baseEnum, "whereAreYou", "");
				
			}
			else {
				base = perceptsBase;

				if(base.getDistance() > this.maxDistanceGive()) {
					this.setHeading(base.getAngle());
					return this.move();
				}
				else {
					this.setIdNextAgentToGive(base.getID());
					return self.give();
				}
			}*/
		}
		else {
			this.setDebugString("Chercher food");
		}


		if(this.isBlocked()) {
			this.randomHeading();
		}

		return this.move(); 

	}
});


var WarExplorer = new Explorer();
